### PR TITLE
Make rendering template in agent client temporary

### PR DESCRIPTION
**Pull request by edwinyyyu**
# TL;DR
A render of the task template in the agent client can persist across multiple nodes. Solution stores the unrendered template in a separate variable and uses it to recover the unrendered template.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

### CODE REVIEW

File: go/tasks/plugins/webapi/agent/plugin.go [hunk 1]
```diff
@@ -101,6 +103,10 @@ func (p Plugin) Create(ctx context.Context, taskCtx webapi.TaskExecutionContextR
 		return nil, nil, err
 	}
 
+	if taskTemplate.GetContainer() != nil {
+		taskTemplate.GetContainer().Args = argTemplate
```
Comment (pingsutw) [inline go/tasks/plugins/webapi/agent/plugin.go:107]: qq: why do we need to revert args here? did you see any error in the propeller?

Comment (edwinyyyu) [reply go/tasks/plugins/webapi/agent/plugin.go:107]: For tasks with the same task function, the first render will persist, causing all prefixes to correspond to the same node. The {{}} templates will have been replaced so later renders will not find anything to replace.

Comment (edwinyyyu) [reply go/tasks/plugins/webapi/agent/plugin.go:107]: This solution was chosen over makeing a copy of the entire taskTemplate. I'm not familiar with if there is a better place to solve this problem.

Comment (pingsutw) [reply go/tasks/plugins/webapi/agent/plugin.go:107]: LGTM, could you add a comment at line 105?

Comment (edwinyyyu) [reply go/tasks/plugins/webapi/agent/plugin.go:108]: Done.

Comment (pingsutw) [reply go/tasks/plugins/webapi/agent/plugin.go:108]: thanks!

**Review by pingsutw — APPROVED** [review]

**Review by hamersaw — APPROVED** [review]

**Pull request merged**